File: src/main/scala/Models/Mamifero.scala
package Models

import Traits.Animal

abstract class Mamifero(protected val nome: String, protected val idade: Int) extends Animal {

  def getNome() = nome
  def getIdade() = idade

  override def toString() = "Nome: " + getNome() + "\n" + "Idade: " + getIdade()

}

File: Java/src/models/Mamifero.java
package models;

import interfaces.Animal;

public abstract class Mamifero implements Animal {

	final protected String nome;
	final protected int idade;
	
	public Mamifero(String nome, int idade) {
		this.idade = idade;
		this.nome = nome;
	}
	
	public int getIdade() {
		return idade;
	}

	public String getNome() {
		return nome;
	}

}
